src/com/projectkorra/projectkorra/storage/Database.java

```diff
@@ -116,6 +116,9 @@ public void run() {
 	 */
 	public ResultSet readQuery(String query) {
 		try {
+                        if(connection == null || connection.isClosed()) {
+                            open();

```

Comment:
This wouldn't work because it is calling the abstract method "open" which does not have any method body. It wouldn't know to call open() from MySQL.java or SQLite.java